guardian/.github
# Add workflow that performs security checks on templates generated by CDK

## What does this change?

Added a workflow that scans CDK-generated templates for security-related misconfigurations. Allows dangerous config to be caught on push, PR or merge, allowing teams to shift left, rather than relying AWS to give feedback after deployment

## How to test

This should work with both yarn and npm, so testing needs to be done against one of each kind of repo.

An npm example is security hq ([test run here](https://github.com/guardian/security-hq/actions/runs/5289740754/jobs/9573038513))
A yarn example is amigo ([test run here](https://github.com/guardian/amigo/actions/runs/5289791763/jobs/9573152386)). This run failed correctly, as the severity threshold was set to low, and it found some low-level issues.

## How can we measure success?

- It's easier for teams to catch insecure configuration in their infrastructure
- We shift left by catching the issues as early as pushing them to a branch (depending on action setup), and are forced to deal with them before they go live if they are blocking PR checks.
- More secure infrastructure across the Guardian
- Increased confidence in deployments

## Have we considered potential risks?

- This may slow teams down if they initially discover several high or critical level configuration issues in their templates. However, if the action is set up on a branch before it's merged, the findings can be fixed along with the introduction of the template, so it will not block merges to main by people working on other areas of the project
